pyproject.toml

```diff
@@ -25,9 +25,12 @@ dependencies = [
     "h5py>=3.6.0",
     "igor2",
     "xarray>=0.20.2",
-    "pint==0.17",
     "pynxtools>=0.3.2",
-    "setuptools>=64.0.1"
+    "setuptools>=64.0.1",
+    "pint==0.17; python_version <= '3.9'",
+    "numpy<2.0.0; python_version <= '3.9'",
+    "pint>=0.24; python_version >= '3.10'",
+    "numpy>=2.0.0; python_version >= '3.10'"
```

Comment:
Do we necessarily need `>2.0.0` for python `3.10`. A lower version might also work, no?

Reply:
If I remember correctly, they had to make pint 0.24 compatible with numpy 2.0.0. It might be that older versions of numpy can cause issues. I saw a table of important packages being updated to support some changes in np==2.0.0

Reply:
This might be of relevance: https://github.com/hgrecco/pint/issues/1974

Reply:
But I think pint 0.24 is still working with numpy < 2.0.0, right? So, it should be okay to numpy>=1.21.2 for py>=3.10.

Reply:
Yes but I think that `numpy<2.0.0` should still work. Even the numpy 2 migration guide suggests to make the changes such that it works also with lower versions. So I think that `pint==0.24` would also work with `numpy<2.0.0`

Reply:
Alright, then I'll leave it as it is. @domna could you please approve if you are happy?

Reply:
Alright. It seems it is nicely backwards compatible. Cool. We can resolve this comment from my end.